lib/screenplay/suppressed_predictions/prediction_suppression_utils.ex

```diff
@@ -0,0 +1,161 @@
+
+  def jfk_umass_child_stop_data,
+    do: [
+      %{
+        route_id: "Red",
+        stop_id: "70085",
+        direction_id: 0,
+        location_id: @jfk_umass_ashmont_location_id
+      },
+      %{
+        route_id: "Red",
+        stop_id: "70086",
+        direction_id: 1,
+        location_id: @jfk_umass_ashmont_location_id
+      },
+      %{
+        route_id: "Red",
+        stop_id: "70095",
+        direction_id: 0,
+        location_id: @jfk_umass_braintree_location_id
+      },
+      %{
+        route_id: "Red",
+        stop_id: "70096",
+        direction_id: 1,
+        location_id: @jfk_umass_braintree_location_id
+      }
+    ]
+
+  @jfk_umass_child_stop_ids ["70085", "70086", "70095", "70096"]
+
+  @spec suppression_type(
+          suppressed_predictions :: [SuppressedPrediction.t()],
+          location_id :: String.t(),
+          route_id :: String.t(),
+          direction_id :: integer()
+        ) :: :terminal | :stop | :none
+
+  def suppression_type(suppressed_predictions, location_id, route_id, direction_id)
+      when is_green_line(route_id) do
+    suppression_type_from_line_data(
+      suppressed_predictions,
+      location_id,
+      "Green",
+      direction_id
+    )
+  end
+
+  def suppression_type(suppressed_predictions, location_id, route_id, direction_id)
+      when is_sl_waterfront(route_id) do
+    suppression_type_from_line_data(
+      suppressed_predictions,
+      location_id,
+      "Silver",
+      direction_id
+    )
+  end
+
+  def suppression_type(suppressed_predictions, location_id, route_id, direction_id) do
+    suppression_type_from_line_data(
+      suppressed_predictions,
+      location_id,
+      route_id,
+      direction_id
+    )
+  end
+
+  def suppression_type(suppressed_predictions, stop_id)
+      when stop_id in @jfk_umass_child_stop_ids do
+    case Enum.find(jfk_umass_child_stop_data(), fn %{stop_id: id} -> id == stop_id end) do
+      %{
+        location_id: @jfk_umass_ashmont_location_id,
+        route_id: route_id,
+        direction_id: direction_id
+      } ->
+        suppression_type_from_line_data(
+          suppressed_predictions,
+          @jfk_umass_ashmont_location_id,
+          route_id,
+          direction_id
+        )
+
+      %{
+        location_id: @jfk_umass_braintree_location_id,
+        route_id: route_id,
+        direction_id: direction_id
+      } ->
+        suppression_type_from_line_data(
+          suppressed_predictions,
+          @jfk_umass_braintree_location_id,
+          route_id,
+          direction_id
+        )
+
+      nil ->
+        :none
+    end
+  end
+
+  defp suppression_type_from_line_data(
+         suppressed_predictions,
+         location_id,
+         route_id,
+         direction_id
+       ) do
+    Enum.find(suppressed_predictions, fn prediction ->
+      prediction.route_id == route_id && prediction.location_id == location_id &&
+        prediction.direction_id == direction_id
```

Comment:
Since we're doing this lookup repeatedly, it would be good to avoid the linear `Enum.find` to make sure the performance stays reasonable. We could pre-create a `MapSet` and then use a simple `in` to do the lookups.